Core/clim-basic/pointer-tracking.lisp

```diff
@@ -172,128 +172,112 @@
 
 
 ;;; DRAG-OUTPUT-RECORD and DRAGGING-OUTPUT.
+(defun make-default-feedback-function (stream output-record erase repaint dx dy)
+  (declare (ignore stream output-record))
+  (lambda (record sheet x0 y0 x y action)
+    (declare (ignore x0 y0))
+    (setf (output-record-position record) (values (+ x dx) (+ y dy)))
+    (ecase action
+      (:erase
+       (if (output-recording-stream-p sheet)
+           (maybe-funcall erase record sheet)
+           (repaint-sheet sheet (bounding-rectangle record))))
+      (:repaint
+       (when repaint
+         (repaint-sheet sheet (bounding-rectangle record))))
+      (:draw
+       (replay-output-record record sheet))
+      (:finish
+       (when (output-recording-stream-p sheet)
+         (stream-add-output-record sheet record))
+       (replay-output-record record sheet)))))
 
-;;; XXX Unresolved issues:
-;;; multiple-window is completely unsupported.
-;;; window-repaint events while dragging.
-
-;;; Fancy double-buffered feedback function
-(defun make-buffered-feedback-function (record finish-on-release erase-final)
-  (multiple-value-bind (record-x record-y)
-      (output-record-position record)
-    (lambda (record stream initial-x initial-y x y event)
-      (flet ((simple-erase ()
-               (when erase-final
-                 (when (output-record-parent record)
-                   (delete-output-record record (output-record-parent record)))
-                 (with-double-buffering
-                     ((stream record) (buffer-rectangle))
-                   (stream-replay stream buffer-rectangle)))))
-        (let ((dx (- record-x initial-x))
-              (dy (- record-y initial-y)))
-          (typecase event
-            (null
-             (setf (output-record-position record) (values (+ dx x) (+ dy y)))
-             (stream-add-output-record stream record)
-             (stream-replay stream record))
-            (pointer-motion-event
-             ;; Don't do an explicit erase. Instead, update the position of the
-             ;; output record and redraw the union of the old and new
-             ;; positions.
-             (with-bounding-rectangle* (old-x1 old-y1 old-x2 old-y2)
-                 record
-               (when (output-record-parent record)
-                 (delete-output-record record (output-record-parent record)))
-               (setf (output-record-position record)
-                     (values (+ dx x) (+  dy y)))
-               (stream-add-output-record stream record)
-               (with-bounding-rectangle* (new-x1 new-y1 new-x2 new-y2)
-                   record
-                 (multiple-value-bind (area-x1 area-y1 area-x2 area-y2)
-                     (values (min old-x1 new-x1) (min old-y1 new-y1)
-                             (max old-x2 new-x2) (max old-y2 new-y2))
-                   (with-double-buffering
```

Comment:
we've changed interpretation of how we call function with feedback. This approach does not fit this model at all. I didn't notice any flicker with existing demos. repaint and draw seems to be fast enough. n.b with-double-buffering may break code, because it swaps mirrors - that's why we couldn't enable xrender rectangles by default. double buffering requires further thought (this hack is not good).

Reply:
OK.